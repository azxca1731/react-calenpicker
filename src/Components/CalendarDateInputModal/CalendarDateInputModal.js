import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const CalendarDateInputModalDiv = styled.div`
  position: absolute;
  z-index: 1000;
  color: ${props => props.theme.fontColor};
  border: 1px solid ${props => props.theme.fontColor};
  background-color: ${props => props.theme.backgroundColor};
  top: -1px;
  left: 0px;
`;

const CalendarDateInputModalHead = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 1px solid ${props => props.theme.fontColor};
  height: ${props => props.size.height.substr(0, props.size.height.length - 2) * 0.15}px;
`;

const CalendarDateInputModalFooter = styled.div`
  border-top: 1px solid ${props => props.theme.fontColor};
  display: flex;
  justify-content: center;
  align-items: center;
  height: ${props => props.size.height.substr(0, props.size.height.length - 2) * 0.15}px;
`;

const CalendarDateInputModalBody = styled.div`
  height: ${props => props.size.height.substr(0, props.size.height.length - 2) * 0.7}px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const CalendarDateInputModalButton = styled.button`
  background-color: ${props => (props.isAccept ? props.theme.acceptColor : props.isDelete ? "red" : props.theme.cancelColor)};
  color: ${props => props.theme.backgroundColor};
  border: 2px solid ${props => props.theme.fontColor};
  margin: 0px 5px;
  padding: 0.5em 1em;
  opacity: 0.7;
  transition: 0.5s;
  cursor: pointer;

  &:hover {
    opacity: 1;
  }
`;

class CalendarDateInputModal extends React.Component {
  constructor(props) {
    super(props);
    const { target } = props;
    this.state = {
      date: target ? target.date : "",
      text: target ? target.text : "",
      isHoliday: target ? target.isHoliday : true
    };
  }

  handleInputDateChange = event => {
    this.setState({
      date: event.target.value
    });
  };

  handleInputTextChange = event => {
    this.setState({
      text: event.target.value
    });
  };

  handleInputIsHoliday = event => {
    this.setState({
      isHoliday: event.target.checked
    });
  };

  handleAddButtonClicked = () => {
    const { target, updateCalendarText, addCalendarText } = this.props;
    const { text, date, isHoliday } = this.state;
    if (target && target.text) {
      updateCalendarText(target, { text, date, isHoliday });
    } else {
      addCalendarText({ text, date, isHoliday });
    }
    this.handleCloseButtonClicked();
  };

  handleDeleteButtonClicked = () => {
    const { target, deleteCalendarText } = this.props;
    const { text, date, isHoliday } = this.state;
    deleteCalendarText(target, { text, date, isHoliday });
    this.handleCloseButtonClicked();
  };

  handleCloseButtonClicked = () => {
    const { handleTargetSetValue, handleModal } = this.props;
    this.setState({
      text: "",
      date: "",
      isHoliday: true
    });
    handleTargetSetValue({
      text: "",
      date: "",
      isHoliday: true
    });
    handleModal(false);
  };

  render() {
    const { size, target } = this.props;
    const { isHoliday, date, text } = this.state;
    return (
      <CalendarDateInputModalDiv style={size}>
        <CalendarDateInputModalHead size={size}>날짜 추가</CalendarDateInputModalHead>
        <CalendarDateInputModalBody size={size}>
          <div>
            <label htmlFor="date">날짜: </label>
            <input id="date" value={date} onChange={this.handleInputDateChange} placeholder="2019-12-10" />
          </div>
          <br />
          <br />
          <div>
            <label htmlFor="text">이름: </label>
            <input id="text" value={text} onChange={this.handleInputTextChange} placeholder="휴가" />
          </div>
          <br />
          <br />
          <div>
            <label htmlFor="isHoliday">휴일 여부: </label>
            <input id="isHoliday" type="checkbox" checked={isHoliday} onChange={this.handleInputIsHoliday} />
          </div>
          <br />
        </CalendarDateInputModalBody>
        <CalendarDateInputModalFooter size={size}>
          <CalendarDateInputModalButton isAccept onClick={this.handleAddButtonClicked}>
            {target && target.text ? "수정" : "추가"}
          </CalendarDateInputModalButton>
          {target && target.text ? (
            <CalendarDateInputModalButton isDelete onClick={this.handleDeleteButtonClicked}>
              삭제
            </CalendarDateInputModalButton>
          ) : null}
          <CalendarDateInputModalButton onClick={this.handleCloseButtonClicked}>닫기</CalendarDateInputModalButton>
        </CalendarDateInputModalFooter>
      </CalendarDateInputModalDiv>
    );
  }
}

CalendarDateInputModal.defaultProps = {
  addCalendarText: () => {},
  handleModal: () => {},
  size: {}
};

CalendarDateInputModal.propTypes = {
  addCalendarText: PropTypes.func,
  handleModal: PropTypes.func,
  handleTargetSetValue: PropTypes.func,
  deleteCalendarText: PropTypes.func,
  updateCalendarText: PropTypes.func,
  size: PropTypes.object,
  target: PropTypes.shape({
    date: PropTypes.string,
    text: PropTypes.string,
    isHoliday: PropTypes.bool
  })
};

export default CalendarDateInputModal;
