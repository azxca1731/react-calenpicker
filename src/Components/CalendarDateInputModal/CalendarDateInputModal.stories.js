import React from "react";
import { ThemeProvider } from "styled-components";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

import light from "Styles/theme/light";
import dark from "Styles/theme/dark";
import CalendarDateInputModal from "Components/CalendarDateInputModal";

storiesOf("Components", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={light}>{story()}</ThemeProvider>)
  .add("CalendarDateInputModal light", () => <CalendarDateInputModal />);

storiesOf("Components", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={dark}>{story()}</ThemeProvider>)
  .add("CalendarDateInputModal Dark", () => <CalendarDateInputModal />);
