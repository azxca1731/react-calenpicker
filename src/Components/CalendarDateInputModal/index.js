import CalendarDateInputModal from "./CalendarDateInputModal";
export default CalendarDateInputModal;
