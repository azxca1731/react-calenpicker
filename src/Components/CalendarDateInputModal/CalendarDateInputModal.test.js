import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import CalendarDateInputModal from "./index";

describe("CalendarDateInputModal", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<CalendarDateInputModal />);
    expect(wrapper).toHaveLength(1);
  });

  it("renders with snapshot", () => {
    const wrapper = shallow(<CalendarDateInputModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
