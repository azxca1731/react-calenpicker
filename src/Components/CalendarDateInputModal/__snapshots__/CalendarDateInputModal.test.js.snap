// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDateInputModal renders with snapshot 1`] = `
<styled.div
  style={
    Object {
      "height": "450px",
      "width": "300px",
    }
  }
>
  <styled.div
    size={
      Object {
        "height": "450px",
        "width": "300px",
      }
    }
  >
    자세히 보기
  </styled.div>
  <styled.div
    size={
      Object {
        "height": "450px",
        "width": "300px",
      }
    }
  >
    <styled.div
      size={
        Object {
          "height": "450px",
          "width": "300px",
        }
      }
    >
      <styled.div>
        아무런 스케줄이 없습니다
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    size={
      Object {
        "height": "450px",
        "width": "300px",
      }
    }
  >
    <styled.button
      onClick={[Function]}
    >
      닫기
    </styled.button>
  </styled.div>
</styled.div>
`;
